frontend/src/static/js/v_authorship.js

```diff
@@ -156,6 +157,36 @@ window.vAuthorship = {
       return 'Total: Blank: ' + this.totalBlankLineCount + ', Non-Blank: '
           + (this.totalLineCount - this.totalBlankLineCount);
     },
+
+    injectText(text, file) {
+      text += `########################################## ${file.path}\n`;
+      file.segments.forEach((seg) => {
+        text += seg.lines.join('\n');
+        text += '\n';
+      });
+      return text;
+    },
+
+    downloadAll() {
+      if (this.filesDownloaded.length === 0) {
+        return;
+      }
+      let text = '';
+      this.filesDownloaded.forEach((file) => {
+        text = this.injectText(text, file);
```

Comment:
name this `injectFile` instead of injectText, cos `injectText` implies that all the formatting for the text is done in there, but the function more for text formatting for each file.

also, it would make more sense to create a `renderFile` function instead that returns the text for each file, then do all the appending in `downloadAll`, instead of passing the text to the function then passing it back out again